File: algorithms/matlab/sorting/quick_sort.m
function sortedArray = quick_sort(arr)
    % Quick Sort implementation in MATLAB
    
    if length(arr) <= 1
        sortedArray = arr;
        return;
    end
    
    pivot = arr(end);
    arr(end) = [];
    
    less = arr(arr < pivot);
    greater = arr(arr >= pivot);
    
    sortedArray = [quick_sort(less), pivot, quick_sort(greater)];
end

% Example Usage:
% arr = [10, 7, 8, 9, 1, 5];
% disp('Original Array:');
% disp(arr);
% sorted = quick_sort(arr);
% disp('Sorted Array:');
% disp(sorted);

File: algorithms/python/sorting/quick_sort.py
def quick_sort(arr):
    """
    Python implementation of Quick Sort Algorithm.
    Time Complexity: O(n log n) average, O(n^2) worst case
    Space Complexity: O(log n)
    """
    if len(arr) <= 1:
        return arr
    else:
        pivot = arr.pop()
        greater = []
        lower = []
        
        for item in arr:
            if item > pivot:
                greater.append(item)
            else:
                lower.append(item)
        
        return quick_sort(lower) + [pivot] + quick_sort(greater)

if __name__ == "__main__":
    test_arr = [10, 7, 8, 9, 1, 5]
    print(f"Original Array: {test_arr}")
    sorted_arr = quick_sort(test_arr)
    print(f"Sorted Array: {sorted_arr}")
